pkg/util/containers/v2/metrics/collector_ecs_fargate_test.go

```diff
@@ -0,0 +1,291 @@
+					return errors.New("container stats not cached")
+				}
+				return nil
+			},
+		},
+		{
+			name: "cache is populated, but invalid",
+			fields: fields{
+				lastScrapeTime: time.Now().Add(-30 * time.Second),
+			},
+			args: args{
+				containerID:   "container_id",
+				cacheValidity: 10 * time.Second,
+				clientFunc:    func(context.Context, string) (*v2.ContainerStats, error) { return &apiStats, nil },
+			},
+			loadFunc: func() { cache.Cache.Set("ecs-stats-container_id", &cachedStats, statsCacheExpiration) },
+			want:     &apiStats,
+			wantErr:  false,
+			wantFunc: func() error {
+				if _, found := cache.Cache.Get("ecs-stats-container_id"); !found {
+					return errors.New("container stats not cached")
+				}
+				return nil
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := &ecsFargateCollector{
+				lastScrapeTime: tt.fields.lastScrapeTime,
+			}
+
+			tt.loadFunc()
+			got, err := e.stats(tt.args.containerID, tt.args.cacheValidity, tt.args.clientFunc)
+
+			assert.Equal(t, tt.wantErr, err != nil)
+			assert.EqualValues(t, tt.want, got)
+			assert.Nil(t, tt.wantFunc())
+
+			cache.Cache.Flush()
+		})
+	}
+}
+
+func TestConvertEcsNetworkStats(t *testing.T) {
+	type args struct {
+		netStats v2.NetStatsMap
+		networks map[string]string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *ContainerNetworkStats
+	}{
+		{
+			name: "nominal case",
+			args: args{
+				netStats: v2.NetStatsMap{"eth1": v2.NetStats{RxBytes: 2398415937, RxPackets: 1898631, TxBytes: 1259037719, TxPackets: 428002}},
+				networks: map[string]string{},
+			},
+			want: &ContainerNetworkStats{
+				Interfaces:  map[string]InterfaceNetStats{"eth1": {BytesRcvd: floatPtr(2398415937), PacketsRcvd: floatPtr(1898631), BytesSent: floatPtr(1259037719), PacketsSent: floatPtr(428002)}},
+				BytesRcvd:   floatPtr(2398415937),
+				PacketsRcvd: floatPtr(1898631),
+				BytesSent:   floatPtr(1259037719),
+				PacketsSent: floatPtr(428002),
+			},
+		},
+		{
+			name: "custom interface name",
+			args: args{
+				netStats: v2.NetStatsMap{"eth1": v2.NetStats{RxBytes: 2398415937, RxPackets: 1898631, TxBytes: 1259037719, TxPackets: 428002}},
+				networks: map[string]string{"eth1": "custom_iface"},
+			},
+			want: &ContainerNetworkStats{
+				Interfaces:  map[string]InterfaceNetStats{"custom_iface": {BytesRcvd: floatPtr(2398415937), PacketsRcvd: floatPtr(1898631), BytesSent: floatPtr(1259037719), PacketsSent: floatPtr(428002)}},
+				BytesRcvd:   floatPtr(2398415937),
+				PacketsRcvd: floatPtr(1898631),
+				BytesSent:   floatPtr(1259037719),
+				PacketsSent: floatPtr(428002),
+			},
+		},
+		{
+			name: "multiple interfaces",
+			args: args{
+				netStats: v2.NetStatsMap{
+					"eth0": v2.NetStats{RxBytes: 2398415937, RxPackets: 1898631, TxBytes: 1259037719, TxPackets: 428002},
+					"eth1": v2.NetStats{TxBytes: 2398415937, TxPackets: 1898631, RxBytes: 1259037719, RxPackets: 428002},
```

Comment:
Nit: Should we choose different values for each interface? Then, we could be sure that we are picking the right values for each interface and not just the first one that we find, for example.